cmake/Modules/FindFLANN.cmake

```diff
@@ -42,16 +42,38 @@ endif()
 
 # First try to locate FLANN using modern config
 find_package(flann NO_MODULE ${FLANN_FIND_VERSION} QUIET)
+
 if(flann_FOUND)
   unset(flann_FOUND)
   set(FLANN_FOUND ON)
+
   # Create interface library that effectively becomes an alias for the appropriate (static/dynamic) imported FLANN target
   add_library(FLANN::FLANN INTERFACE IMPORTED)
-  if(FLANN_USE_STATIC)
+
+  if(TARGET flann::flann_cpp_s AND TARGET flann::flann_cpp)
+    if(PCL_FLANN_REQUIRED_TYPE MATCHES "DONTCARE")
```

Comment:
This is confusing, because it ignores the PCL_FLANN_REQUIRED_TYPE if only 1 type of flann is present.

Is that the expected behavior?

Reply:
Its just setting what version is found. If both are found I check for which one is requested and set accordingly. If no specific version is requested, follow if we build PCL as SHARED or STATIC.

If only one type is found, set it to that type. Then in the main CMakeLists it is checked if the requested build type is the correct one found.